### PR TITLE
feat(sdk-core): add logic to handle ERC20 tokens for staking

### PR DESCRIPTION
**Pull request by noel-bitgo**
BOS-495

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] My code compiles correctly for both Node and Browser environments
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My commits follow [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) and I have properly described any BREAKING CHANGES
- [ ] The ticket or github issue was included in the commit message as a reference
- [ ] I have made corresponding changes to the documentation and on any new/updated functions and/or methods - [jsdoc](https://jsdoc.app/)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: modules/sdk-core/src/bitgo/staking/stakingWallet.ts [hunk 1]
```diff
@@ -35,7 +35,9 @@ export class StakingWallet implements IStakingWallet {
   }
 
   get coin(): string {
-    return this.wallet.coin();
+    return (this.wallet.baseCoin as any).hasOwnProperty('tokenConfig')
```
Comment (mmcshinsky-bitgo) [inline modules/sdk-core/src/bitgo/staking/stakingWallet.ts:38]: What are all the possibilities of `any` here. Can we define this?

Comment (noel-bitgo) [reply modules/sdk-core/src/bitgo/staking/stakingWallet.ts:38]: it's `ERC20Token` defined in sdk-coin-eth, but if I use that we would have a circular dependencies between sdk-core and sdk-coin-eth.

do you recommend to have a type define in sdk-core with just the `tokenConfig` in the baseCoin?

Comment (bitgoAaron) [reply modules/sdk-core/src/bitgo/staking/stakingWallet.ts:38]: I would suggest adding `tokenConfig?: BaseTokenConfig` to `IBaseCoin` instead.

Comment (noel-bitgo) [reply modules/sdk-core/src/bitgo/staking/stakingWallet.ts]: done, didn't know we had this in the sdk-core 👍

**Pull request merged**